### PR TITLE
Generate basic licenses json

### LINKED ISSUE #84: Generate licenses json
eyeing this - https://github.com/hierynomus/license-gradle-plugin

Goal would be to generate the licenses json that's used by apollo at runtime, so hopefully we can have this library generate the top level data and then scrape the VCS info out of it

### PR DESCRIPTION
**Pull request by ZacSweers**
Unfortunately a lot of POMs don't actually include (github) VCS info, so this isn't as useful as I was hoping it would be.

TODO: generate this into assets somewhere, differentiate which ones are still missing.

Resolves #84

### CODE REVIEW

**Comment by ZacSweers** [comment]
Idea - put in a mapping of known artifacts to their github repos for ones that are just missing poms in the code itself, this way it doesn't get out of date easily

File: buildSrc/src/main/kotlin/dev/zacsweers/catchup/gradle/LicensesJsonGenerator.kt [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+  private fun Configuration.artifactView(): ArtifactView {
+    return incoming.artifactView {
+      attributes {
+        attribute(Attribute.of("artifactType", String::class.java), "android-classes")
+      }
+    }
+  }
+}
+
+open class LicensesJsonGenerator @Inject constructor(objects: ObjectFactory) : DefaultTask() {
```
Comment (autonomousapps) [inline buildSrc/src/main/kotlin/dev/zacsweers/catchup/gradle/LicensesJsonGenerator.kt]: I'm not sure where the docs are on this (but they do exist). You can instead declare this class as
```kotlin
abstract class LicensesJsonGenerator : DefaultTask()
```
salient points:
1. it's abstract instead of open
2. It doesn't define any constructor

With that, you can then do this for your properties:
```kotlin
@get:Classpath
abstract val artifactFiles: ConfigurableFileCollection
```
salient points:
1. the property is abstract
2. you no longer need to initialize it manually from ObjectFactory

This works because Gradle extends all this stuff at runtime. E.g., it'll create a class named `LicensesJsonGenerator_Decorated`. It uses that decorated version.

Basically, Gradle already "knows" how to initialize a `ConfigurableFileCollection`. It can do automatically what you've done manually. Benefit is a bit less boilerplate when defining your task classes.

Please note that this is fine and you should leave it as-is:
```kotlin
  @get:Internal
  lateinit var configuration: Configuration
```

Comment (ZacSweers) [reply buildSrc/src/main/kotlin/dev/zacsweers/catchup/gradle/LicensesJsonGenerator.kt]: good shout, done here f2253a7
